src/SHA.jl

```diff
@@ -51,7 +51,7 @@ for (f, ctx) in [(:sha1, :SHA1_CTX),
         # open("test.txt") do f
         #     sha256(f)
         # done
-        $f(io::IO) = $f(readbytes(io))
+        $f(io::IO) = @compat $f(read(io))
```

Comment:
this syntax parses and evaluates fine on earlier julia, so it's not a syntax rewrite and shouldn't need the compat macro.